pkg/pillar/cmd/nodeagent/nodeagent.go

```diff
@@ -577,10 +588,19 @@ func handleLastRebootReason(ctx *nodeagentContext) {
 	// if reboot stack size crosses max size, truncate
 	if len(rebootStack) > maxJSONAttributeSize {
 		runes := bytes.Runes([]byte(rebootStack))
-		if len(runes) > maxJSONAttributeSize {
-			runes = runes[:maxRebootStackSize]
+		len := len(runes)
+		// Repeat the check for runes
+		if len > maxJSONAttributeSize {
+			// Get the tail of the stack, because stack grows down and in
+			// case of truncation it is important to see the beginning of
+			// it (bottom) rather then the end (top).
```

Comment:
Any idea how likely this tail of dmesg will miss the stack trace part?

Reply:
dmesg contains the actual crash backtrace at the bottom, then we jump into the crash kernel. So in order to pick up the stack and not the beginning of the booting procedure it makes sense to keep the tail, not the head.

Reply:
So it always the case. But this stack we use (if I'm not mistaken) only to publish on the cloud node page. So it is not very much important (just overall understanding that something nasty happened), since we eventually will go to the logs.  But here I can be mistaken.

Reply:
Yeach we sort of have it in two places, so it depends whether one wants to look for events and patters looking at the device info sent to the controller, the logs, or both. But what you have is fine.